src/algopy_testing/op.py

```diff
@@ -866,13 +866,13 @@ def get_asset_param(a: algopy.Asset | algopy.UInt64 | int) -> tuple[typing.Any,
             if not active_txn:
                 raise ValueError("No active transaction found to reference asset")
 
-            asset_id = a.value if isinstance(a, (algopy.Asset)) else int(a)
-            asset_data = active_txn.assets[asset_id]
+            asset_id = int(a.id) if isinstance(a, (algopy.Asset)) else int(a)
```

Comment:
Something to consider with all these ops that take an available Application, Account or Asset, when the parameter is a `UInt64` or `int`, it could be referring to an Id (in the case of Applications and Assets) OR it could be the offset into the appropriate foreign array of the transaction (Account, Application and Asset). I don't think the testing context currently provides a way to define these?

We should either provide a way to map an foriegn array index to an application/account/asset
OR at a minimum raise an error if the value looks to be an index (e.g. below the starting ID values of 1001)

Reply:
@daniel-makerx added this to AssetParamsGet, AcctParamsGet and AppParamsGet, the AssetHolding get seems to only work with actual indexes not indexes of an asset in foreign refs. Feel free to reopen if you think some other ops are missing this